rdkcentral/rdkservices
# Cherry-pick changes from 22Q4 sprint

- RDK-36951: Support use of paths without further modifications

Reason for change: Backward compatibiliy with already deployed systems
Test Procedure: Verify plugin configuration is correctly set Risks: Small

Signed-off-by: Filipe Norte <filipe.norte@sky.uk>

- RDK-36108: Add JSPP support

- [WebKitBrowserPlugin] Add API to execute JS in web page context

- And initial support for cloud cookie jar

- WebKitBrowser: Fix cookie crypto jar linkage

Signed-off-by: Ievgen Mutavchi <Ievgen_Mutavchi@comcast.com>

- RDK-36108: Set JSPP environment variables

- Implement SetCookies and RefreshCookieJar APIs for WebKit browser plugin.

- Use GLib WebKitCookieManager APIs (...get/set_cookie_jar)

RDK-38159: TimeZone update

Reason for change: Update TZ from file (updated by external component) Test Procedure: None
Priority: P1
Risks: None

Signed-off-by: Vivek.A <vivek_arumugam@comcast.com>

Move impl to separate file

Rearrange TZ support files to be used by Extension

Address review comments
* Remove UtilsStrings.h dependency
* Dereference GFile & GFileMonitor instances

Add WebKit User Content Filter support for WebKit plugin

Add "contentfilter" config for WebKitBrowser plugin that is used as JSON with user contetn filter

Only GLIB based WebKit API supports this feature

Redirect logs to another systemd service.

With rdkbrowser2 all browser logs were part of a parent service. Webkit browser plugin prints logs as wpeframework service.

This change adds possibility to redirect all webkit browser plugin logs (stdout and stderr) to another systemd service to preserve any logging monitors and file watchers still working.

RDK-38353: Enable MSAA Cairo compositor by default

Reason for change: Make cairo compositor configurable for Html & Lightning callsigns
Test Procedure: Refer ticket
Priority: P1
Risks: None

Signed-off-by: Vivek.A <vivek_arumugam@comcast.com>

RDK-38406: Startupscreen launch failed with webkitbrowserplugin backend

Reason for change: On startup screen launch the "Internet" precondition is not met for HtmlApp callsign. This precondition is not required for HtmlApp & LightningApp callsigns as it might be possible for launching locally hosted pages with those callsigns
Test Procedure: Verify startupscreen is loaded with webkitbrowserplugin Priority: P1
Risks: None

Signed-off-by: Vivek.A <vivek_arumugam@comcast.com>

[WebKitBrowser] initialize UserAgent string

[WebKitBrowsr] fix handling of canceled loads

XIONE-10935: Failing to launch Crackle app

Reason for change: Crackle app fails when FileSystem API
 is present. With https://bugs.webkit.org/show_bug.cgi?id=202796
 the DirectoryUpload feature was enabled for WPE port. Disabling
 it to keep the behavior same as wpe-2.22 and there are no use
 cases with that
Test Procedure: None
Priority: P1
Risks: None

Signed-off-by: Vivek.A <vivek_arumugam@comcast.com>

DELIA-58272 : Log error code in webkitbrowser plugin when URL fails to load

Reason for change: Log error code in webkitbrowser plugin when URL fails to load Test Procedure: Specified in ticket
Risks: NA

Signed-off-by: Simi Mathew <simim@tataelxsi.co.in>
